File: server/app/models/services/service.py
from typing import Optional, TYPE_CHECKING, List
from sqlmodel import SQLModel, Field, Relationship

if TYPE_CHECKING:
    from .action import Action
    from .reaction import Reaction
    from ..users.user_service import UserService

class Service(SQLModel, table=True):
    __tablename__ = "service"
    id: int = Field(default=None, primary_key=True)
    name: str = Field(index=True, unique=True, nullable=False)
    description: Optional[str] = None
    image_url: str
    category: str
    color: str = "#000000"
    oauth_required: bool

    actions: List["Action"] = Relationship(back_populates="service", sa_relationship_kwargs={"cascade": "all, delete-orphan"})
    reactions: List["Reaction"] = Relationship(back_populates="service", sa_relationship_kwargs={"cascade": "all, delete-orphan"})
    users: List["UserService"] = Relationship(back_populates="service", sa_relationship_kwargs={"cascade": "all, delete-orphan"})

File: frontend/app/types/service.ts
/*
 ** EPITECH PROJECT, 2025
 ** Area_Mirroring
 ** File description:
 ** service
 */

import { ConfigRespAct } from "./config";

export interface Act {
  id: number
  name: string
  description: string
}

export interface ActDetails {
  id: number,
  act: Act,
  config: ConfigRespAct[],
}

export interface Service {
  id: number
  name: string
  image_url: string
  category: string
  color: string
}

export interface SpecificService {
  id: number
  name: string
  description: string
  image_url: string
  category: string
  color: string
  oauth_required: boolean
}

export interface OAuth_login {
  name: string
  color: string
  image_url: string
}
